web/client/epics/usersession.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import Rx from 'rxjs';
+import { error } from '../actions/notifications';
+import { SAVE_USER_SESSION, userSessionSaved, loading, saveUserSession } from "../actions/usersession";
+import { createResource, createCategory, updateResource } from '../api/persistence';
+import {userSelector} from '../selectors/security';
+import { wrapStartStop } from '../observables/epics';
+import isString from "lodash/isString";
+
+const saveUserSessionErrorStatusToMessage = (status) => {
+    switch (status) {
+    default:
+        return 'userSession.saveErrorNotification.defaultMessage';
+    }
+};
+
+/**
+ * Returns a user session save epic.
+ * The epic triggers on a SAVE_USER_SESSION action.
+ *
+ * @param {function} nameSelector selector that builds the session identifier
+ * @param {*} sessionSelector selector that builds the session object to save
+ * @param {*} idSelecotr selector of the identifier for the current session (if it already exists)
+ * @example
+ *
+ * const nameSelector = (state) => ({state.context.name + "." + state.security.user.name})
+ * const sessionSelector = (state) => ({state.map.present})
+ * const idSelector = (state) => ({state.usersession.id})
+ * const epic = saveUserSessionEpic(nameSelector, sessionSelector, idSelector)
+ * setInterval(() => dispatch({type: SAVE_USER_SESSION}), 60 * 1000)
+ */
+export const saveUserSessionEpic = (nameSelector, sessionSelector, idSelector = () => {}) => (action$, store) => action$
```

Comment:
because they are not epics, maybe a different naming, like `saveSessionEpicCreator` is more descriptive... or any better idea, if you can think any

Reply:
saveSessionEpicCreator is fine